### PR TITLE
Multigrid: Add subbn3d

### PR DESCRIPTION
**Pull request by SuX97**
Add subbn3d.

### CODE REVIEW

File: mmaction/core/evaluation/eval_hooks.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 from mmcv.runner import Hook
 from torch.utils.data import DataLoader
 
+from mmaction.models.common.sub_batch_bn3d import SubBatchBN3d
```
Comment (dreamerlin) [inline mmaction/core/evaluation/eval_hooks.py:8]:
```
from mmaction.models import SubBatchBN3d
```

File: mmaction/models/common/sub_batch_bn3d.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            self.affine = False
+
+    def _get_aggregated_mean_std(self, means, stds, n):
+        """Calculate the aggregated mean and stds.
+
+        Args:
+            means (tensor): mean values.
+            stds (tensor): standard deviations.
+            n (int): number of sets of means and stds.
+        """
```
Comment (dreamerlin) [inline mmaction/models/common/sub_batch_bn3d.py:38]:
```
Returns:
   tuple[torch.Tensor, torch.Tensor]: (mean, std). mean is xxxx. std is xxxx.
```

File: mmaction/models/common/sub_batch_bn3d.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            self.weight = torch.nn.Parameter(torch.ones(self.num_features))
+            self.bias = torch.nn.Parameter(torch.zeros(self.num_features))
+        else:
+            self.affine = False
+
+    def _get_aggregated_mean_std(self, means, stds, n):
+        """Calculate the aggregated mean and stds.
+
+        Args:
+            means (tensor): mean values.
```
Comment (dreamerlin) [inline mmaction/models/common/sub_batch_bn3d.py:35]: torch.Tensor

File: mmaction/models/common/sub_batch_bn3d.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+    def aggregate_stats(self):
+        """Synchronize running_mean, and running_var.
+
+        Call this before eval.
+        """
+        if self.split_bn.track_running_stats:
+            (
+                self.bn.running_mean.data,
+                self.bn.running_var.data,
```
Comment (dreamerlin) [inline mmaction/models/common/sub_batch_bn3d.py:53]: the last comma should be removed.

File: mmaction/models/common/sub_batch_bn3d.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+            ) = self._get_aggregated_mean_std(
+                self.split_bn.running_mean,
+                self.split_bn.running_var,
+                self.num_splits,
```
Comment (dreamerlin) [inline mmaction/models/common/sub_batch_bn3d.py:57]: The last comma should be removed.

File: mmaction/models/heads/base.py [hunk 6]
```diff
@@ -3,7 +3,7 @@
 import torch
 import torch.nn as nn
 
-from ...core import top_k_accuracy
+from ...core.evaluation.accuracy import top_k_accuracy
```
Comment (dreamerlin) [inline mmaction/models/heads/base.py:6]: why?

Comment (SuX97) [reply mmaction/models/heads/base.py]: That confuses me too. Check [this CI log](https://github.com/open-mmlab/mmaction2/pull/133/checks?check_run_id=993420247#step:10:40) out. The only difference is [this](https://github.com/open-mmlab/mmaction2/pull/133/commits/18343a55a99a79dda45e1af9674e0cd876ea5c55).

Comment (SuX97) [reply mmaction/models/heads/base.py]: I guess it is what's called 'circular import'. When import all from `.evaluate`, it imports `eval_hook`. But in eval_hook it imports `mmaction.models`, which imports `head.py` in which it imports `evaluate`. Forms a circle!

Comment (dreamerlin) [reply mmaction/models/heads/base.py]: It was caused by a circular importing. You can try to move it inside `def after_train_epoch(self, runner):` (like `from mmaction.apis import single_gpu_test`)

File: mmaction/models/common/sub_batch_bn3d.py [hunk 7]
```diff
@@ -0,0 +1,74 @@
+import torch
+import torch.nn as nn
+from mmcv.cnn import NORM_LAYERS
+
+
+@NORM_LAYERS.register_module()
+class SubBatchBN3d(nn.Module):
```
Comment (dreamerlin) [inline mmaction/models/common/sub_batch_bn3d.py:7]: Need docstring

File: mmaction/models/heads/base.py (hunk 6 above)
Comment (SuX97) [reply mmaction/models/heads/base.py]: So I either write `from ...core.evaluation.accuracy import top_k_accuracy`, or in eval_hooks.py write `from mmaction.models.common import SubBatchBN3d` to avoid importing head.py. Both ugly.

Comment (SuX97) [reply mmaction/models/heads/base.py]: Moving it inside `aggregate_sub_bn_stats` will cause repeatly importing it, hurting the efficiency. But it will only run for like 53 times in i3d, is it acceptable?

Comment (dreamerlin) [reply mmaction/models/heads/base.py]: Yeah, I think it is acceptable.

**Review by dreamerlin — APPROVED** [review]
